cmake/boards.cmake

```diff
@@ -0,0 +1,123 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Function to list architectures and boards
+# Takes the board root as a semicolon-separated list in:
+# BOARD_ROOT
+# Populates the following variables in parent scope:
+# arch_list
+# boards_for_${arch}
+
+# Additional arguments:
+# print Print the board list with
+# file_out Filename to save the boards list in
+# indent Prepend an additional indendation string to each line
+function(list_archs_boards print file_out indent)
```

Comment:
This function is doing too many things.

There should be two functions:

```
list_archs  (print file_out indent arch_list_output_variable)
list_boards(print file_out indent arch board_list_output_variable)
```


one function for determining the number of arch's.

And another for populating the boards for an

Reply:
Addressed